# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/rnnoise/hotfix-arm-neon.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

src/vec_neon.h:35:10: fatal error: os_support.h: No such file or directory
   35 | #include "os_support.h"
      |          ^~~~~~~~~~~~~~

--- rnnoise-0.2/src/vec_neon.h.vanilla	2025-06-09 11:50:49.029785343 +0200
+++ rnnoise-0.2/src/vec_neon.h	2025-06-09 11:51:02.550346900 +0200
@@ -32,7 +32,11 @@
 #define VEC_NEON_H
 
 #include <arm_neon.h>
-#include "os_support.h"
+
+/** Set n elements of dst to zero */
+#ifndef OVERRIDE_OPUS_CLEAR
+#define OPUS_CLEAR(dst, n) (memset((dst), 0, (n)*sizeof(*(dst))))
+#endif
 
 #if defined(__arm__) && !defined(__aarch64__) && (__ARM_ARCH < 8 || !defined(__clang__))
 /* Emulate vcvtnq_s32_f32() for ARMv7 Neon. */
